### PR TITLE
Upgrade amazon-states-language-service dependency to version 1.4.0

### PR DESCRIPTION
**Pull request by yoodan93**
## Description

Step Functions is launching new features for Amazon States Language, and we are updating the [amazon-states-language-service](https://github.com/aws/amazon-states-language-service) version in the toolkit to enable the new features.

## Motivation and Context

We are releasing the new Amazon States Language features on 8/13, we would like to have the linter in the toolkit for Step Functions to have the new features available.

## Related Issue(s)

No related issues.

## Testing

Ran tests locally on MAC

## Checklist

- [x] I have read the **README** document
- [x] I have read the **CONTRIBUTING** document
- [x] My code follows the code style of this project
- [x] I have added tests to cover my changes
- [x] All new and existing tests passed
- [x] A short description of the change has been added to the changelog using the script `npm run newChange`

## License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: .changes/next-release/Feature-ed68db23-5ec8-42ba-8a31-8dfd4a826f92.json [hunk 1]
```diff
@@ -0,0 +1,4 @@
+{
+    "type": "Feature",
+    "description": "Upgrade amazon-states-language-service dependency version to 1.4.0"
```
Comment (awschristou) [inline .changes/next-release/Feature-ed68db23-5ec8-42ba-8a31-8dfd4a826f92.json:3]: This description doesn't have meaning to customers. What will be different for them when they obtain the version of the toolkit containing this changelog entry?

Comment (yoodan93) [reply .changes/next-release/Feature-ed68db23-5ec8-42ba-8a31-8dfd4a826f92.json:3]: Will change it to something more meaningful for the customer

Comment (justinmk3) [reply .changes/next-release/Feature-ed68db23-5ec8-42ba-8a31-8dfd4a826f92.json:3]: this old json file needs to be deleted too

Comment (yoodan93) [reply .changes/next-release/Feature-ed68db23-5ec8-42ba-8a31-8dfd4a826f92.json]: it has been removed :)

**Pull request merged**